### PR TITLE
Attribute Parser Should return an error if Email is longer than 80 characters

### PR DESCRIPTION
**Pull request by MiguelAHM**

### CODE REVIEW

File: whois-rpsl/src/main/java/net/ripe/db/whois/common/rpsl/AttributeParser.java [hunk 1]
```diff
@@ -199,6 +199,10 @@ public InternetAddress parse(final String s) {
                 final String address = parsed[0].getAddress();
                 final String localPart = address.substring(0, address.indexOf('@'));
 
+                if (address.length() > 80){
```
Comment (eshryane) [inline whois-rpsl/src/main/java/net/ripe/db/whois/common/rpsl/AttributeParser.java:202]: Define 80 in a static field like `MAXIMUM_LENGTH` to the EmailParser class.

File: whois-rpsl/src/test/java/net/ripe/db/whois/common/rpsl/AttributeSyntaxTest.java [hunk 2]
```diff
@@ -244,7 +244,8 @@ public void email() {
         verifySuccess(ObjectType.PERSON, AttributeType.E_MAIL, "user@host.org");
         verifySuccess(ObjectType.PERSON, AttributeType.E_MAIL, "Any User <user@host.org>");
         verifySuccess(ObjectType.PERSON, AttributeType.E_MAIL, "a@a.a");
-        verifySuccess(ObjectType.PERSON, AttributeType.E_MAIL, "0@2.45678901234567890123456789012345678901234567890123456789012345678901234567890");
+        verifyFailure(ObjectType.PERSON, AttributeType.E_MAIL, "0@2.45678901234567890123456789012345678901234567890123456789012345678901234567890"); //To large email
```
Comment (eshryane) [inline whois-rpsl/src/test/java/net/ripe/db/whois/common/rpsl/AttributeSyntaxTest.java:247]: Be consistent and group the success and failure cases together in that method

File: whois-rpsl/src/main/java/net/ripe/db/whois/common/rpsl/AttributeParser.java (hunk 1 above)
Comment (MiguelAHM) [reply whois-rpsl/src/main/java/net/ripe/db/whois/common/rpsl/AttributeParser.java]: Done

File: whois-api/src/test/java/net/ripe/db/whois/api/mail/dequeue/BouncedMessageParserTest.java [hunk 3]
```diff
@@ -87,6 +87,14 @@ public void parse_permanent_delivery_failure_message_rfc822_headers_real() throw
         assertThat(bouncedMessage.emailAddresses(), containsInAnyOrder("testing4@ripe.net"));
     }
 
+    @Test
+    public void parse_permanent_delivery_failure_to_a_too_big_recipient_then_recipient_ignored() throws Exception {
```
[the remark this answers is not in the document]
Comment (MiguelAHM) [reply whois-api/src/test/java/net/ripe/db/whois/api/mail/dequeue/BouncedMessageParserTest.java]: Done

File: whois-rpsl/src/test/java/net/ripe/db/whois/common/rpsl/AttributeSyntaxTest.java (hunk 2 above)
Comment (MiguelAHM) [reply whois-rpsl/src/test/java/net/ripe/db/whois/common/rpsl/AttributeSyntaxTest.java]: Done

File: whois-rpsl/src/main/java/net/ripe/db/whois/common/rpsl/AttributeParser.java [hunk 4]
```diff
@@ -180,6 +180,8 @@ public String parse(final String s) {
 
     final class EmailParser implements AttributeParser<InternetAddress> {
 
+        public static final int MAXIMUM_LENGTH = 80;
```
Comment (eshryane) [inline whois-rpsl/src/main/java/net/ripe/db/whois/common/rpsl/AttributeParser.java:183]: Make private

File: whois-rpsl/src/main/java/net/ripe/db/whois/common/rpsl/AttributeParser.java [hunk 5]
```diff
@@ -199,6 +201,10 @@ public InternetAddress parse(final String s) {
                 final String address = parsed[0].getAddress();
                 final String localPart = address.substring(0, address.indexOf('@'));
 
+                if (address.length() > MAXIMUM_LENGTH){
+                    throw new AttributeParseException("Illegal address length, maximum supported length is 80 (%s)", s);
```
Comment (eshryane) [inline whois-rpsl/src/main/java/net/ripe/db/whois/common/rpsl/AttributeParser.java:205]: Don't hard-code 80 in message

File: whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/BouncedMessageParser.java [hunk 6]
```diff
@@ -183,11 +186,23 @@ private List<String> extractRecipients(final DeliveryStatus deliveryStatus) {
                 LOGGER.error("Wrong formatted recipient {}", recipient);
                 continue;
             }
-            recipients.add(finalRecipientMatcher.group(2));
+            final String email = finalRecipientMatcher.group(2);
+            if (isValidEmail(email)) {
+                recipients.add(email);
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/BouncedMessageParser.java:191]: Double-check the impact of an empty recipient in parser response (if the email is not valid)

Comment (eshryane) [reply whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/BouncedMessageParser.java:191]: Checked code, it's OK if no email address in `parse()` response, message will still be deleted.

However please add a test for this (i.e. parse a bounce response with-too long email).

Comment (MiguelAHM) [reply whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/BouncedMessageParser.java:191]: I added a test called 
permanent_delivery_failure_to_one_recipient_too_long_address_then_delete_message_without_marking_undeliverable_address
in MailBounceTestIntegration
can you double check? This should take care that the message is not still there and that the email is not stored in database

File: whois-rpsl/src/main/java/net/ripe/db/whois/common/rpsl/AttributeParser.java (hunk 5 above)
Comment (MiguelAHM) [reply whois-rpsl/src/main/java/net/ripe/db/whois/common/rpsl/AttributeParser.java]: Done

File: whois-rpsl/src/main/java/net/ripe/db/whois/common/rpsl/AttributeParser.java (hunk 4 above)
Comment (MiguelAHM) [reply whois-rpsl/src/main/java/net/ripe/db/whois/common/rpsl/AttributeParser.java]: Done

**Comment by MiguelAHM** [comment]
Test ok
https://gitlab.ripe.net/swe-database-team/whois/-/pipelines/627720

**Review by eshryane — APPROVED** [review]

**Pull request merged**